gmusicapi/protocol/musicmanager.py

```diff
@@ -147,15 +148,28 @@ def get_track_clientid(filepath):
         # converting sum to base64
         # removing trailing ===
 
-        # My implementation is _not_ the same hash the music manager will send;
-        # they strip tags first. But files are differentiated across accounts,
-        # so this shouldn't cause problems.
+        m = hashlib.md5()
+        ext = os.path.splitext(filepath)[1]
 
-        # This will attempt to reupload files if their tags change.
+        # delete=False is needed because the NamedTemporaryFile
+        # can't be opened by name a second time on Windows otherwise.
+        temp = NamedTemporaryFile(suffix=ext, delete=False)
 
-        m = hashlib.md5()
         with open(filepath, 'rb') as f:
-            m.update(f.read())
+            temp.write(f.read())
+
+            audio = mutagen.File(temp.name, easy=True)
+            audio.delete()
+            audio.save()
+
+            m.update(temp.read())
+
+        # Delete the NamedTemporaryFile.
+        temp.close()
```

Comment:
Yeah, I was thinking something like:
```python

with open(filepath, 'rb') as f, NamedTemporaryFile... as temp:
    try:
        ...
    finally:
        os.remove(temp.name)
```

> Putting something that could potentially throw an exception in a finally clause isn't a great idea.

Because a previous exception could get swallowed? We could add an except + log.exception to make sure it would get logged -- I don't think there's any callers expecting to handle failures in that block anyway.